app/design/frontend/Elogic/maksym/Magento_Catalog/web/js/qty-counter.js

```diff
@@ -0,0 +1,30 @@
+define(["jquery"], function ($) {
+    ("use strict");
+
+    $(function () {
+        $("#adds").on("click", add);
+        $("#subs").on("click", remove);
+    });
+
+    function add() {
+        var input = $("input#qty"),
+            value = input.val();
+
+        input.val(++value);
+
+        if (value > 0) {
+            $("#subs").removeAttr("disabled");
+        }
+    }
+
+    function remove() {
+        var input = $("input#qty"),
+            value = input.val();
+
+        if (value > 1) {
+            input.val(--value);
+        } else {
+            $("#subs").attr("disabled", "disabled");
+        }
+    }
+});
```

Comment:
Looks good 👍🏻 

But it should work as a jQuery widget, so you could reuse it in different places (cart page, mini cart etc). The problem is that you couldn't use this JS when you have more than one qty counter on a page, because you use IDs (#qty, #adds, #subs). So your JS will work only ones per page.

There should be a possibility to init your widget on a specific element, but not on a "*" selector, like you've done in app/design/frontend/Elogic/maksym/Magento_Catalog/templates/product/view/addtocart.phtml

It will be something like:
```
<div class="control" data-mage-init='{"qtyControls":{}}'>
         <button type="button" class="subs"></button>
         <input type="number" name="qty" id="qty" 
                     class="qty" min="0" value="<?= $block->getProductDefaultQty() * 1 ?>"
                     title="<?= $block->escapeHtmlAttr(__('Qty')) ?>" class="input-text qty"
                     data-validate="<?= $block->escapeHtml(json_encode($block->getQuantityValidators())) ?>" />
          <button type="button" class="adds"></button>
</div>
```

In this case, you could work with increase/decrease buttons and input using something like this:
```
$(this.element).find('.subs');
$(this.element).find('.adds');
$(this.element).find('.qty');
```

`$(this.element)` is an element on which you init your widget, so it's `<div class="control">` in this case

Check this article about widget codding standards (I'll update the ticket with the correct link):
https://developer.adobe.com/commerce/php/coding-standards/jquery-widgets/

Reply:
noted / done

Reply:
Your JS is still just a JS code, you need to declare a widget:
$.widget('maksym. qtyControls', {
       // your code 
});